### ISSUE TITLE
Store: Split wc-api-dev into two plugins

### ISSUE THREAD

**Issue by justinshreve** [Automattic/wp-calypso#18603]
We have been adding a lot of additional code to `wc-api-dev` allowing us to do custom stuff on the PHP side. It's time to split it into two plugins so we can keep API code destined for WC core and other Calypso/Store on .com code separate & organized.

**Comment by justinshreve** [comment]
Something to keep in mind: We should have a utility function for getting asset paths in the new plugin. The path will be different when testing as a installed `plugin` vs  being pulled in by `composer`. Example: https://github.com/woocommerce/wc-api-dev/pull/61

**Comment by timmyc** [comment]
Another utility function will be to get the proper site slug from the domain to use when linking back to calypso pages (https://github.com/woocommerce/wc-api-dev/blob/master/hotfixes/wc-api-dev-masterbar-menu.php#L23-L27)

**Comment by timmyc** [comment]
@justinshreve I was going to start working on this task - any preferences or ideas on the naming of this new plugin? Also should it live in the `woocommerce` or `automattic` org? I have permissions to create a new repo in automattic but not woo fwiw.

**Comment by justinshreve** [comment]
Let's go with the `automattic` repo. We chose `woocommerce` for `wc-api-dev` because the original goal of that plugin was to be a staging area for future API versions and much more core focused (until we started needing to use it for other reasons). This plugin feels more Automattic/Calypso specific.

In terms of name, I'm not sure, but I'm open to suggestions. Last time we talked about it I think I mentioned `wc-calypso-bridge` or something.
